### PR TITLE
NFC: cleanup manifest constructor defaults

### PR DESCRIPTION
**Pull request by tomerd**
motivation: safer code

changes:
* remove nil defaults from manfiest constructor
* update tests to use the test oriented helpers that set the defaults for testing

### CODE REVIEW

**Comment by tomerd** [comment]
ran into this when working on another fix, so brought out as separate PR

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
@swift-ci smoke test

**Review by MaxDesiatov — APPROVED** [review]
Thanks for cleaning this up!

**Comment by tomerd** [comment]
@swift-ci smoke test

File: Sources/PackageModel/Manifest/Manifest.swift [hunk 1]
```diff
@@ -106,16 +105,16 @@ public final class Manifest: Sendable {
         path: AbsolutePath,
         packageKind: PackageReference.Kind,
         packageLocation: String,
-        defaultLocalization: String? = nil,
+        defaultLocalization: String?,
         platforms: [PlatformDescription],
-        version: TSCUtility.Version? = nil,
-        revision: String? = nil,
+        version: TSCUtility.Version?,
+        revision: String?,
```
Comment (neonichu) [inline Sources/PackageModel/Manifest/Manifest.swift:111]: Are version and revision mutually exclusive?

**Comment by neonichu** [comment]
IMO, these PRs that combine large formatting changes and code changes are very hard to review. I really think we should either leave the formatting alone or do one pass across the entire codebase, neither formatting nor history preservation are more important than being able to do code reviews.

File: Sources/PackageModel/Manifest/Manifest.swift (hunk 1 above)
Comment (tomerd) [reply Sources/PackageModel/Manifest/Manifest.swift:111]: not that I know. Am I missing something?

**Comment by tomerd** [comment]
> IMO, these PRs that combine large formatting changes and code changes are very hard to review. I really think we should either leave the formatting alone or do one pass across the entire codebase, neither formatting nor history preservation are more important than being able to do code reviews.

in this case its actually not formatting changes that made the PR large. there was a large set of call sites to update. but if there is consensus to do a one-tilme formatting PR across the entire repo, we can do that too.

**Comment by MaxDesiatov** [comment]
I don't a single combined formatting PR, but maybe I'd prefer that to happen after the soundness script runs on CI for every PR so that we don't have formatting regressions and a necessity for any follow-up formatting and soundness fixes.

**Comment by tomerd** [comment]
that is a good point. @shahmishal do you think you can help us set up such CI job? basically a job that just runs the Utilities/soundness script

**Pull request merged**